include/aikido/control/BarrettFingerKinematicSimulationPositionCommandExecutor.hpp

```diff
@@ -8,67 +8,70 @@
 #include <future>
 #include <mutex>
 #include <condition_variable>
+#include <chrono>
 
 namespace aikido {
 namespace control {
 
-/// This executor mimics the behavior of BarretFinger. 
-/// It moves a finger to a desired point; it may stop early if 
+/// This executor mimics the behavior of BarretFinger.
+/// It moves a finger to a desired point; it may stop early if
 /// joint limit is reached or collision is detected.
 /// Only the proximal joint is actuated; the distal joint moves with mimic ratio.
 /// When collision is detected on the distal link, the finger stops.
 /// When collision is detected on the proximal link, the distal link moves
 /// until it reaches joint limit or until distal collision is detected.
-class BarrettFingerPositionCommandExecutor
+class BarrettFingerKinematicSimulationPositionCommandExecutor
 {
 public:
   /// Constructor.
-  /// \param _finger Finger to be controlled by this Executor.
-  /// \param _proximal Index of proximal dof
-  /// \param _distal Index of distal dof
-  /// \param _collisionDetector CollisionDetector for detecting self collision
-  ///        and collision with objects. 
-  /// \param _collisionOptions Default is (enableContact=false, binaryCheck=true,
-  ///        maxNumContacts = 1.) 
-  ///        See dart/collison/Option.h for more information 
-  BarrettFingerPositionCommandExecutor(
-    ::dart::dynamics::ChainPtr _finger, int _proximal, int _distal,
-    ::dart::collision::CollisionDetectorPtr _collisionDetector,
-    ::dart::collision::CollisionOption _collisionOptions
+  /// \param[in] finger Finger to be controlled by this Executor.
+  /// \param[in] proximal Index of proximal dof
+  /// \param[in] distal Index of distal dof
+  /// \param[in] collideWith CollisionGroup to check collision with fingers.
+  /// \param[in] collisionOptions Default is (enableContact=false, binaryCheck=true,
+  ///        maxNumContacts = 1.)
+  ///        See dart/collison/Option.h for more information
+  BarrettFingerKinematicSimulationPositionCommandExecutor(
+    ::dart::dynamics::ChainPtr finger, size_t proximal, size_t distal,
+    ::dart::collision::CollisionGroupPtr collideWith,
```

Comment:
My understanding is that the arguments should be `finger`, `proximal`, `distal` (which are all required arguments), `collisionDetector` and `collideWith` (which are both optional and default to `nullptr`), and `collisionOptions` (which remains optional and defaults to `false` and `1`).

It seems like the three cases are:
1. **If `collisionDetector` and `collideWith` are not specified**, set `collisionDetector ` to an `FCLCollisionDetector` and `collideWith` to a new empty `CollisionGroup` with that collision detector.
2. **If `collisionDetector` is specified and `collideWith` is not**, set `collideWith` to a new empty `CollisionGroup` with `collisionDetector` as the collision detector
3. **If `collisionDetector` and `collideWith` are specified**, check that the collision detector for `collideWith` matches `collisionDetector`. if not, set `collideWith` to a new empty `CollisionGroup` with `collisionDetector` as the collision detector

@jslee02 Is my understanding correct and if so, does e16880e9fca951199c50b7639a8ba46f7dbaa318 seem reasonable? In particular, I'm unsure what it means for two collision detectors to not be the same. Do they have to be literally the same object or just the same type of collision detector? It also seems surprising to me to automatically set `collideWith` to an empty collision group if there were originally collision objects in there, so I think I might be misunderstanding something. :confused: